src/transformState.js

```diff
@@ -6,6 +6,29 @@
  */
 function transformState(state, actions) {
   // write code here
+  const resObj = state;
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        Object.assign(resObj, actions[i].extraData);
+        break;
+
+      case 'removeProperties':
+        for (const ch of actions[i].keysToRemove) {
+          delete resObj[ch];
+        }
+        break;
+
+      case 'clear':
+        for (const q in resObj) {
```

Comment:
Use 'Object.keys(state).forEach(key => delete state[key])' instead of 'for in' loop to clear properties.